deps/src/ideals.cpp

```diff
@@ -713,10 +713,12 @@ void singular_define_ideals(jlcxx::Module & Singular)
   Singular.method("scHilb", [](ideal I, ring r, jlcxx::ArrayRef<int> a) {
     const ring origin = currRing;
     rChangeCurrRing(r);
-    intvec * v = hFirstSeries(I, NULL, r->qideal);
-    int *    content = v->ivGetVec();
+    bigintmat * v = hFirstSeries0b(I, r->qideal, NULL,NULL, r,coeffs_BIGINT);
     for (int j = 0; j < v->length(); j++)
-      a.push_back(content[j]);
+    {
+      number n=(*v)[j];
+      a.push_back(n_Int(n,coeffs_BIGINT));
```

Comment:
So if `n` does not fit into an `int`, what happens then? Is an exception thrown (where is it caught), or does it again just silently overflow?

Can't we just return an array of bigints? That requires changing `a` to `ArrayRef<jl_value_t>`, I think, and then putting `n_Z` or `BigInt` values in there?

Reply:
Singulkar prints an error message and return 0, if the conversion to int fails